library/src/main/java/eu/inmite/android/lib/validations/form/FieldAdapterFactory.java

```diff
@@ -49,6 +51,11 @@ public static IFieldAdapter getAdapterForField(View view) {
 				sJoinedAdapter = new JoinedAdapter();
 			}
 			adapter = sJoinedAdapter;
+		} else if (annotation != null && FieldsEqual.class.equals(annotation.annotationType())) {
```

Comment:
I think you need to distinguish between FieldEqualsAdapter and JoinedAdapter. You need to have separate fields for them. Otherwise, you would end up with one or another forever.